testing/go/regression/reader/reader.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"unicode"
+)
+
+const filepath = "/path/to/expected/"  // This should point to the "expected" folder, and end with a forward slash
+const outtestpath = "/path/to/output/" // This should point to wherever test files should be saved, and end with a forward slash
+
+func main() {
+	names := []string{
+		"advisory_lock.out",
```

Comment:
I think all else being equal, I would rather check these files into source control and implement a harness / runner to run them directly, rather than having a utility to translate them into Go code. Translating these files into Go code doesn't really add anything, it's just a different format (and another thing to slow down compile and startup time). It also will make it harder for us to track changes to these tests as they change upstream.